### ISSUE TITLE
Indexing pressure does not account for updates which increase the size of documents

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#97818]
Today the indexing pressure tracker accounts for the sizes of update operations received by the primary, but these could be tiny in comparison with the size of the resulting document. However we do need to hold the resulting document in memory on the primary while replicating, so we must notify the indexing pressure tracker of any increase in size resulting from an update operation.

We've seen cases where a node can have orders of magnitude more indexing traffic in memory than is tracked by the indexing pressure mechanism, which means the node does not push back on indexing properly and in extreme cases may experience an OOME.

**Comment by serenachou** [comment]
Do we have an updated status on this ticket @DaveCTurner ?

**Comment by DaveCTurner** [comment]
I'm not in the relevant team so cannot answer this authoritatively, but I believe it's up to date.

**Comment by kingherc** [comment]
I believe this issue is true also for cases of scripted updates, where the script may be updating a document (e.g., adding values to a field) and the resulting document / source (e.g., as constructed in `org.elasticsearch.action.update.UpdateHelper#prepareUpdateScriptRequest()` function) may be large and we do not account for it. Correct me if I'm wrong.

**Comment by DaveCTurner** [comment]
> this issue is true also for cases of scripted updates

Yes, I didn't intend to imply that this case wasn't included. All updates may increase the size of the document in an untracked fashion, whether they involve scripts or otherwise; indeed with a scripted update there is almost unbounded scope for this size increase.

**Comment by DaveCTurner** [comment]
I don't think #146247 completes this so I'm reopening it.

**Comment by blerer** [comment]
https://github.com/elastic/elasticsearch/issues/97819 increase the index pressure based on the size of the prepared update.
